### PR TITLE
feat: adds audience.

### PR DESCRIPTION
**Pull request by atulep**
Adds support for GDCH credentials.

### CODE REVIEW

**Review by arithmetic1728 — APPROVED** [review]
LGTM

File: gapic/templates/tests/unit/gapic/%name_%version/%sub/test_%service.py.j2 [hunk 1]
```diff
@@ -902,6 +929,29 @@ def test_{{ service.name|snake_case }}_transport_auth_adc(transport_class):
         )
 
 
+{% if 'grpc' in opts.transport %}
```
Comment (atulep) [inline gapic/templates/tests/unit/gapic/%name_%version/%sub/test_%service.py.j2:932]: Remove this check

**Review by viacheslav-rostovtsev — APPROVED** [review]
LGTM, but only merge after auth library merges their `with_gdch_audience` part

**Comment by viacheslav-rostovtsev** [comment]
Auth was released so this is fine to release as well.

**Pull request merged**